src/arrays/arraymath.jl

```diff
@@ -34,29 +34,53 @@ function *{B<:OrthonormalBasis}(op::DualMatrix{B}, ket::QuVector{B})
 end
 
 # ket * bra -> operator
-function *{B<:OrthonormalBasis}(ket::QuVector{B}, bra::DualVector{B}) 
-    return QuArray(A_mul_Bc(rawcoeffs(ket), rawcoeffs(bra)), 
-                   bases(ket,1), 
+function *{B<:OrthonormalBasis}(ket::QuVector{B}, bra::DualVector{B})
+    return QuArray(A_mul_Bc(rawcoeffs(ket), rawcoeffs(bra)),
+                   bases(ket,1),
                    bases(bra,1))
 end
 
 # operator * operator -> operator
 function *{B<:OrthonormalBasis}(qm::QuMatrix{B}, dm::DualMatrix{B})
-    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)), 
-                   bases(qm,1), 
+    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)),
+                   bases(qm,1),
                    bases(dm,2))
 end
 
 function *{B<:OrthonormalBasis}(dm::DualMatrix{B}, qm::QuMatrix{B})
-    return QuArray(Ac_mul_B(rawcoeffs(dm), rawcoeffs(qm)), 
-                   bases(dm,1), 
+    return QuArray(Ac_mul_B(rawcoeffs(dm), rawcoeffs(qm)),
+                   bases(dm,1),
                    bases(qm,2))
 end
 
-function *{B<:OrthonormalBasis}(qm1::QuMatrix{B}, qm2::QuMatrix{B}) 
-    return QuArray(rawcoeffs(qm1)*rawcoeffs(qm2), 
-                   bases(qm1,1), 
+function *{B<:OrthonormalBasis}(qm1::QuMatrix{B}, qm2::QuMatrix{B})
+    return QuArray(rawcoeffs(qm1)*rawcoeffs(qm2),
+                   bases(qm1,1),
                    bases(qm2,2))
 end
 
 *(dm1::DualMatrix, dm2::DualMatrix) = (dm1.qarr*dm2.qarr)'
+
+# scaling
+function *(num::Number, qarr::QuArray)
+    return QuArray(num*qarr.coeffs, qarr.bases)
+end
+
+function norm(qarr::QuArray)
+    if ndims(rawcoeffs(qarr)) == 1
+        return norm(rawcoeffs(qarr))
+    else
+        return vecnorm(rawcoeffs(qarr))
+    end
+end
+
+function normalize(qarr::QuArray)
+    return normalize!(copy(qarr))
+end
+
+function normalize!(qarr::QuArray)
+    for i in 1:length(qarr.coeffs)
+        qarr.coeffs[i] = qarr.coeffs[i]/norm(qarr)
```

Comment:
Might be more efficient(?) to use `scale!(norm(qarr), qarr.coeffs)`. For some container types this will for example call a BLAS routine which is more optimized/accurate.